supervisor/addons/model.py

```diff
@@ -511,22 +515,30 @@ def url(self) -> str | None:
     @property
     def with_icon(self) -> bool:
         """Return True if an icon exists."""
-        return self.path_icon.exists()
+        if self._path_icon_exists is None:
+            self._path_icon_exists = self.path_icon.exists()
```

Comment:
In the end this still will do I/O in the asyncio event loop. :cry: 

We had a similar problem when making check_port async, with the dynamic ingress port assignment in #4677.

I think we could make `_check_ingress_port()` more generic (e.g. `_check_addon()`), and update the these things there in an executor. Thoughts? :thinking:

Reply:
I mean we could just make `_clear_cache` into `_refresh_cache` and call it during `load` too

Reply:
Ok I decided to go with the `refresh_cache` model, where the value is always cached and we update it any time the repo does. As noted when we talked there's a risk with this since it means we'll redo our existence checks for all addons in a repo anytime the repo updates. But hopefully that shouldn't be too bad compared to doing this stat check in the event loop for every addon anytime we need to show a list of addons.

The other reason I settled on this model as opposed to making the python properties into methods that run in the executor is because then each `with_<x>` method would run as a separate task. With a `refresh_cache` option I can do all 4 of these at once. It seemed inefficient to do it as 4 separate ones and I didn't want every consumer to have to know the caching details.

I'm still open to other ideas though.

Reply:
Actually I guess I'm kind of double caching these. Really the installed addons should just look at the store to see the value and we cache it there. I'll see if that works

Reply:
Yeah I think combining the update make sense instead of having 4 tasks. I see you went the "reuse store if possible" route :+1: